File: CS/TokenEditCustomDropDown/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraEditors;
using DevExpress.XtraTreeList;

namespace TokenEditCustomDropDown {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            CustomDropDown dropDownGallery = new CustomDropDown();
            tokenEdit1.Properties.CustomDropDownControl = new CustomDropDown();
        }

        private void tokenEdit1_ValidateToken(object sender, DevExpress.XtraEditors.TokenEditValidateTokenEventArgs e) {
            e.IsValid = true;
        }
    }
}

File: VB/TokenEditCustomDropDown/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraEditors
Imports DevExpress.XtraTreeList

Namespace TokenEditCustomDropDown
    Partial Public Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            Dim dropDownGallery As New CustomDropDown()
            tokenEdit1.Properties.CustomDropDownControl = New CustomDropDown()
        End Sub

        Private Sub tokenEdit1_ValidateToken(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.TokenEditValidateTokenEventArgs) Handles tokenEdit1.ValidateToken
            e.IsValid = True
        End Sub
    End Class
End Namespace
